service/project.go

```diff
@@ -150,6 +153,26 @@ func (uis *UIServer) modifyProject(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	for i, pv := range responseRef.PatchVariants {
+		if strings.TrimSpace(pv.Variant) == "" {
+			uis.LoggedError(w, r, http.StatusBadRequest, errors.Errorf("variant regex #%d can't be empty string", i+1))
+			return
+		}
+		if strings.TrimSpace(pv.Task) == "" {
+			uis.LoggedError(w, r, http.StatusBadRequest, errors.Errorf("task regex #%d can't be empty string", i+1))
+			return
+		}
+
+		if _, err := regexp.Compile(pv.Variant); err != nil {
+			uis.LoggedError(w, r, http.StatusBadRequest, errors.Wrapf(err, "variant regex #%d is invalid", i+1))
+			return
+		}
+		if _, err := regexp.Compile(pv.Task); err != nil {
+			uis.LoggedError(w, r, http.StatusBadRequest, errors.Wrapf(err, "task regex #%d is invalid", i+1))
+			return
+		}
```

Comment:
should we try and compile all of them so that it returns all errors, rather than only the first?

Reply:
Probably, that would be cleaner to the user, especially since we can't do accurate client-side verification. Will do.